src/main/java/resources/NasaResource.java

```diff
@@ -0,0 +1,72 @@
+package resources;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.Status;
+import millom.sandbox.CustomException.InvalidWeatherException;
+import millom.sandbox.service.WeatherService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/nasaapi")
+public class NasaResource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NasaResource.class);
+
+  @Inject
+  private WeatherService weatherService;
+
+  @Path("/mars/weather/")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getMarsWeather(@QueryParam("feed") String feed,
+      @QueryParam("feedtype") String feedType,
+      @QueryParam("version") float version, @QueryParam("category") String category) {
+
+    if (feed == null || feed.length() == 0 || !feed.equals("weather")) {
+      return getErrorResponse("feed");
+    }
+    if (feedType == null || feedType.length() == 0 || !feedType.matches("json|xml|text")) {
```

Comment:
Practically, the code now can only parse JSON responses. I'd remove all other options (they don't and can't work), and hardcode feed type.

Reply:
Added these for practice purpose. Now removing other formats and hardcoded to json.
```
if (feedType == null || feedType.length() == 0 || !feedType.equals("json")) {
      return getErrorResponse("feedType");
    }
```